invenio/legacy/bibindex/engine.py

```diff
@@ -98,6 +98,7 @@
     deserialize_via_marshal
 
 from sqlalchemy.exc import DatabaseError
+from MySQLdb import IntegrityError
```

Comment:
I've checked, it doesn't work. Probably because as you can see below, the exception comes from the legacy function `run_sql`, and not SQLalchemy.